improver_tests/utilities/test_normalise_to_reference.py

```diff
@@ -0,0 +1,305 @@
+    snow_data = 0.1 * 0.4 * np.ones(shape, dtype=np.float32)
+
+    for data in [rain_data, sleet_data, snow_data]:
+        data[1, :, :] = 2 * data[1, :, :]
+
+    rain_cube = set_up_probability_cube(
+        rain_data, thresholds, variable_name="rainfall_rate", threshold_units="m s-1"
+    )
+    sleet_cube = set_up_probability_cube(
+        sleet_data,
+        thresholds,
+        variable_name="lwe_sleetfall_rate",
+        threshold_units="m s-1",
+    )
+    snow_cube = set_up_probability_cube(
+        snow_data,
+        thresholds,
+        variable_name="lwe_snowfall_rate",
+        threshold_units="m s-1",
+    )
+
+    return iris.cube.CubeList([rain_cube, sleet_cube, snow_cube])
+
+
+def test_percentile_basic(
+    input_percentile_cubes, reference_percentile_cube, expected_percentile_cubes
+):
+    """Test basic usage that the input cubes are updated correctly."""
+    output = normalise_to_reference(input_percentile_cubes, reference_percentile_cube)
+    output_sum = output[0].data + output[1].data + output[2].data
+
+    assert output == expected_percentile_cubes
+    assert np.array_equal(output_sum, reference_percentile_cube.data)
+
+
+def test_probability_basic(
+    input_probability_cubes, reference_probability_cube, expected_probability_cubes
+):
+    """Test basic usage that the input cubes are updated correctly."""
+    output = normalise_to_reference(input_probability_cubes, reference_probability_cube)
+    output_sum = output[0].data + output[1].data + output[2].data
+
+    assert output == expected_probability_cubes
+    assert np.array_equal(output_sum, reference_probability_cube.data)
+
+
+@pytest.mark.parametrize("ignore_zero_total", (True, False))
+def test_zero_total(
+    input_percentile_cubes,
+    reference_percentile_cube,
+    expected_percentile_cubes,
+    ignore_zero_total,
+):
+    """Test cubes are updated correctly when some values in input_cubes are zero in
+    all input cubes.
+    """
+    for index in range(len(input_percentile_cubes)):
+        input_percentile_cubes[index].data[0, :, :] = 0.0
+        expected_percentile_cubes[index].data[0, :, :] = 0.0
+
+    if not ignore_zero_total:
+        with pytest.raises(
+            ValueError, match="There are instances where the total of input"
+        ):
+            normalise_to_reference(
+                input_percentile_cubes, reference_percentile_cube, ignore_zero_total
+            )
+    else:
+        output = normalise_to_reference(
+            input_percentile_cubes, reference_percentile_cube, ignore_zero_total
+        )
+        output_sum = output[0].data + output[1].data + output[2].data
+
+        reference_with_zeroes = reference_percentile_cube.copy()
+        reference_with_zeroes.data[0, :, :] = 0.0
+
+        assert output == expected_percentile_cubes
+        assert np.array_equal(output_sum, reference_with_zeroes.data)
+
+
+def test_single_input_cube(input_percentile_cubes, reference_percentile_cube):
```

Comment:
I'm not sure why we would ever do this. Perhaps it should raise an exception instead?
Also - add a test for the inputs being an empty cube list.

Reply:
Added an exception for when `len(cubes) < 2' and modified this test to look for that exception instead